mRemoteNG/mRemoteNG
# Use async HttpClient instead of obsolete WebClient

## Description

- Corrected the warnings related to **WebClient** by replacing it with **System.Net.Http**.
- Rewritten a lot of code in AppUpdater.cs to support HttpClient and removed legacy or unused code (e.g. Event-based Asynchronous Pattern (EAP)) and replaced it with APM (Asynchronous Programming Model). Unfortunately, there were no unit tests for HTTP layer, so I haven't added mine.
- Several event handlers were converted from "public void" to "public async void", where other used methods return "async Task" or "async string/int".
- I have also removed some region directives, because they are empty now. I can return them back if needed.

## Motivation and Context
Removes obsolete WebClient. Wanted to help the project.

## How Has This Been Tested?
Manually:
1. Change the assembly version to something old;
2. Start the application and either download the new update, or check whether the new update is available.

## Screenshots (if appropriate):

## Types of changes
- [x] Changed feature (non-breaking change which changes functionality)
- [ ] Changed feature (**breaking** change which changes functionality)
- [ ] Updated translation

## Checklist:
- [ ] All Tests within VisualStudio are passing
